src/contract.rs

```diff
@@ -89,35 +100,106 @@ pub fn add_file<S: Storage, A: Api, Q: Querier>(
     time: u64,
     content: Vec<u8>,
     mode: String,
-) -> StdResult<HandleResponse> {
-    // let sender_address_raw = deps.api.canonical_address(&env.message.sender)?;
-    // config(&mut deps.storage).update(|mut state| {
-    //     if sender_address_raw != state.owner {
-    //         return Err(StdError::Unauthorized { backtrace: None });
-    //     }
-    //     state.count = count;
-    //     Ok(state)
-    // })?;
-    // debug_print("count reset successfully");
-    // Ok(HandleResponse::default())
-    let callback = HandleResponse::default();
+) -> StdResult<HandleAnswer> {
+    let sender_address_raw = deps.api.canonical_address(&env.message.sender)?;
+
+    let block = Block::<DefaultParams>::encode(
+        DagCborCodec,
+        Code::Sha2_256,
+        &ipld!({
+            "owner": sender_address_raw.to_string(),
+            "path": path,
+            "type": content_type,
+            "content": content,
+            "time": time,
+        }),
+    )
+    .unwrap();
+
+    let cid = block.cid().to_string();
+    let data = block.data().to_vec();
 
+    let callback = HandleAnswer::AddFile { cid: cid.clone() };
+    let id = cid.into_bytes();
+    save_to_store(&mut deps.storage, &id, &data)?;
     Ok(callback)
 }
 
-//pub fn query<S: Storage, A: Api, Q: Querier>(
-//    deps: &Extern<S, A, Q>,
-//    msg: QueryMsg,
-//) -> StdResult<Binary> {
-    // match msg {
-    //     QueryMsg::GetCount {} => to_binary(&query_count(deps)?),
-    // }
-//}
-
-//fn query_count<S: Storage, A: Api, Q: Querier>(deps: &Extern<S, A, Q>) -> StdResult<CountResponse> {
-    // let state = config_read(&deps.storage).load()?;
-    // Ok(CountResponse { count: state.count })
-//}
+fn get_metadata<S: Storage, A: Api, Q: Querier>(
+    deps: &Extern<S, A, Q>,
+    cid: String,
+) -> StdResult<QueryAnswer> {
+    let try_cid = Cid::new_v0(
+        libipld::cid::multihash::MultihashGeneric::from_bytes(&cid.into_bytes()).unwrap(),
+    )
+    .unwrap();
+
+    if try_cid.codec() > 0 {
+        panic!("AAAaaaaa!!!!");
```

Comment:
add err message, "Invalid CID". next iteration find out how to return errors in cosmwasm